spec/models/order_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe Order, type: :model do
+  let(:user) { create(:user) }
+  let(:cart) { create(:cart, user: user) }
+  let(:book) { create(:book) }
+  let(:line_item) { create(:line_item, cart: cart, book: book) }
+  let(:order) { create(:order, user: user) }
+
+  describe "#purchase_line_items" do
+    it "assigns cart's line items to order when order is created" do
+      order.purchase_line_items
```

Comment:
You can use the following syntax to explicitly check that the `purchase_line_items` action is causing the changes that you want:

```
expect { order.purchase_line_items }.to change { order.reload.line_items }.to include(line_item)
expect { order.purchase_line_items }.to change { cart.reload.line_items }.to be_empty
```

It will run the code each time so you'd have to split it into two examples, but this is really useful in making sure you didn't accidentally have an empty cart or an order with that line item when the test started.